### PR TITLE
Flatten Entity Schema

### PR DESCRIPTION
**Pull request by BenjaminMoe**
This is a work in progress PR to act as a proposal to address issue https://github.com/w3c-ccg/traceability-vocab/issues/395. This PR is open to track progress. Please do not comment on this PR until it is marked as ready. Discussion can take place in the issue. 

- [x] Move existing `Entity.yml` to `EntityCompat.yml`
- [x] Write script to re-issue credentials for all VC's
- [x] Add a new `Entity.yml` file with updated flattened definition
- [x] Port JSON examples from `EntityCompat` to flattened `Entity`
- [x] Regenerate Traceability Presentation Proof
- [x] Create a script that generates new proofs for all credentials
- [x] Remove `EntityCompat.yml` schema 

`Entity.yml`

```
$linkedData:
  term: Entity
  '@id': https://w3id.org/traceability#Entity
title: Entity
description: A person or organization
type: object
properties:
  type:
    oneOf:
      - type: array
        items:
          type: string
          enum:
            - Entity
      - type: string
        const: Entity
  entityType:
    description: The type of entity
    type: string
    enum:
      - Person
      - Organization
  name:
    title: Name
    description: Name of the organization.
    type: string
    $linkedData:
      term: name
      '@id': https://schema.org/name
  legalName:
    title: Legal Name
    description: The official name of the organization, i.e., the registered company name.
    type: string
    $linkedData:
      term: legalName
      '@id': https://schema.org/legalName
  url:
    title: URL
    description: URL of the person or organization.
    type: string
    $linkedData:
      term: url
      '@id': https://schema.org/url
  taxId:
    title: Tax Identification Number
    description: >-
      The Tax / Fiscal ID of the organization or person, e.g., the TIN in the US
      or the CIF/NIF in Spain.
    type: string
    $linkedData:
      term: taxId
      '@id': https://schema.org/taxID
  address:
    title: Postal Address
    description: The postal address for an organization or place.
    $ref: ./PostalAddress.yml
    $linkedData:
      term: address
      '@id': https://schema.org/PostalAddress
  email:
    title: Email Address
    description: Contact email address.
    type: string
    $linkedData:
      term: email
      '@id': https://schema.org/email
  phoneNumber:
    title: Phone Number
    description: Contact phone number.
    type: string
    $linkedData:
      term: phoneNumber
      '@id': https://schema.org/telephone
required:
  - type
  - entityType
  - name
additionalProperties: true
example: |-
  {
    "type": "Entity",
    "entityType" : "Person",
    "name" : "Kane Heller",
    "firstName": "Kane",
    "lastName": "Heller",
    "email": "Laury96@example.org",
    "phoneNumber": "555-998-3940",
    "worksFor": {
      "type": "Organization",
      "name": "Hyatt - Spencer",
      "description": "Cross-platform 24/7 interface",
      "address": {
        "type": "PostalAddress",
        "streetAddress": "705 Hyman Streets",
        "addressLocality": "Evelinemouth",
        "addressRegion": "Nevada",
        "postalCode": "06775-7181",
        "addressCountry": "Colombia"
      },
      "email": "Bernhard81@example.org",
      "phoneNumber": "555-265-4610",
      "faxNumber": "555-151-9530"
    },
    "jobTitle": "Lead Division Producer"
  }
```

### CODE REVIEW

**Comment by mprorock** [comment]
@rhofvendahl take a look and work with Benjamin to make sure this works for us

File: docs/openapi/components/schemas/common/MillTestReportCertificate.yml [hunk 1]
```diff
@@ -190,75 +148,16 @@ example: |-
...
-              "property": {
-                "type": [
-                  "MechanicalProperty"
-                ],
-                "identifier": "ISO 1352",
-                "name": "Torque-controlled fatigue testing",
-                "description": "ISO 1352:2011 specifies the conditions for performing torsional, constant-amplitude, nominally elastic stress fatigue tests on metallic specimens without deliberately introducing stress concentrations. The tests are carried out at ambient temperature (ideally at between 10 °C and 35 °C) in air by applying a pure couple to the specimen about its longitudinal axis."
+                "identifier": "ISO 180",
+                "name": "Izod Impact Strength Test",
+                "description": "ISO 180 defines the method used for pendulums to determine the impact resistance of a plastic specimen when supported in a cantilever configuration. Test results are used to evaluate the resilience of materials, typically plastics."
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/MillTestReportCertificate.yml:153]: As the `materials` are `typically` (not always) `plastics`, the `specimens` shouldn't be described as only `plastic`.
```suggestion
                "description": "ISO 180 defines the method used for pendulums to determine the impact resistance of a specimen when supported in a cantilever configuration. Test results are used to evaluate the resilience of materials, typically plastics."
```

Comment (BenjaminMoe) [reply docs/openapi/components/schemas/common/MillTestReportCertificate.yml]: Updated in `MillTestReport.yml` and `MillTestReportCertificate.yml`

File: docs/openapi/components/schemas/common/AgParcelDelivery.yml [hunk 2]
```diff
@@ -216,13 +216,7 @@ example: |-
...
       "faxNumber": "555-497-2527"
     },
-    "AgPackage": [
-      {
-        "type": [
-          "AgPackage"
-        ]
-      }
-    ],
+    "AgPackage": [],
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgParcelDelivery.yml:219]: this kind of cleanup is very helpful

File: packages/traceability-schemas/scripts/regenerate.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  });
+
+  if (!verified) {
+    throw new Error('what the effff?');
+  }
+
+  return verifiableCredential;
+};
+
+const main = async () => {
```
Comment (OR13) [inline packages/traceability-schemas/scripts/regenerate.js:61]: clever

**Review by OR13 — APPROVED** [review]

File: docs/openapi/components/schemas/common/TraceablePresentation.yml [hunk 4]
```diff
@@ -33,57 +31,82 @@ example: |-
...
+            "target": "https://example.com/organizations/example-org/presentations/available",
+            "linkRelationship": "OrganizationPresentationEndpoint"
+          }
+        ],
+        "issuanceDate": "2019-12-11T03:50:55Z",
+        "issuer": {
+          "id": "did:key:z6MktHQo3fRRohk44dsbE76CuiTpBmyMWq2VVjvV6aBSeE3U",
+          "type": "Entity",
+          "name": "Glover - Gleason",
+          "legalName": "Glover and Gleason, Llc.",
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/TraceablePresentation.yml:58]:
```suggestion
          "legalName": "Glover and Gleason, LLC",
```

File: docs/openapi/components/schemas/common/TraceablePresentation.yml [hunk 5]
```diff
@@ -33,57 +31,82 @@ example: |-
...
+          "phoneNumber": "555-758-8926",
+          "faxNumber": "555-248-4575",
+          "taxId": "789-56-123",
+          "url": "glover-gleason.example.net",
+          "entityType": "Organization"
+        },
+        "credentialSubject": {
+          "type": "Entity",
+          "name": "Glover - Gleason",
+          "legalName": "Glover and Gleason, Llc.",
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/TraceablePresentation.yml:78]:
```suggestion
          "legalName": "Glover and Gleason, LLC",
```

File: docs/openapi/components/schemas/common/TraceablePresentation.yml (hunk 4 above)
Comment (BenjaminMoe) [reply docs/openapi/components/schemas/common/TraceablePresentation.yml]: Updated

File: docs/openapi/components/schemas/common/TraceablePresentation.yml (hunk 5 above)
Comment (BenjaminMoe) [reply docs/openapi/components/schemas/common/TraceablePresentation.yml]: Updated in presentation and vbc.

**Review by nissimsan — APPROVED** [review]

File: docs/openapi/components/schemas/common/Event.yml [hunk 6]
```diff
@@ -83,10 +83,11 @@ example: |-
         },
         "email": "Sigurd.Langosh@example.com",
         "phoneNumber": "555-296-7144",
-        "faxNumber": "555-149-6658"
+        "faxNumber": "555-149-6658",
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/Event.yml:86]:
```suggestion
        "faxNumber": "+1-555-149-6658",
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Pull request merged**